backend/report_submission/test_views.py

```diff
@@ -758,3 +763,105 @@ def test_post_validates_general_information(self):
             "GSA_MIGRATION not permitted outside of migrations",
             response.context["errors"],
         )
+
+
+def add_session_to_request(request):
+    """Middleware to add session to request."""
+    middleware = SessionMiddleware(lambda x: x)
+    middleware.process_request(request)
+    request.session.save()
+
+
+def add_messages_to_request(request):
+    """Attach a message storage to the request."""
+    setattr(request, "_messages", FallbackStorage(request))
+
+
+class DeleteFileViewTest(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.user = User.objects.create_user(
+            username="user", password="password"
+        )  # nosec
+        self.view = DeleteFileView()
+        self.report_id = "12345"
+        self.path_name = "delete-audit-findings"
+        self.url = reverse(
+            "report_submission:audit-findings-text",
+            kwargs={"report_id": self.report_id},
+        )
+
+    def make_request(self, method="get"):
+        if method.lower() == "post":
+            request = self.factory.post(self.url)
+        else:
+            request = self.factory.get(self.url)
+        request.user = self.user
+        request.path = "/delete/" + self.path_name + "/"
+        add_session_to_request(request)
+        add_messages_to_request(request)
+        return request
+
+    def test_get_no_access(self):
+        request = self.make_request("get")
+        with patch("audit.models.SingleAuditChecklist.objects.get") as mock_get:
+            mock_get.side_effect = SingleAuditChecklist.DoesNotExist
+            with self.assertRaises(PermissionDenied):
+                self.view.get(request, report_id=self.report_id)
+
+    def test_get_successful_render(self):
+        sac = MagicMock(report_id=self.report_id)
```

Comment:
what are the tradeoffs here between using `MagicMock` and `Baker` (which we use in most of our test code)?

Reply:
Baker could be used, but I did not find it necessary here as the focus is more on testing the behavior of the view rendering, so the database interaction was mocked. Using Baker would become relevant if it were necessary to test the database interaction with the Django test database.